espnet2/asr/frontend/s3prl.py

```diff
@@ -82,12 +87,26 @@ def _get_upstream(self, frontend_conf):
         ]:
             s3prl_upstream.model.encoder.layerdrop = 0.0
 
+            # check if adapter is added
```

Comment:
@sw005320 @simpleoier this section takes care of first checking if the upstream selected is wav2vec2 before checking for adapter config and adding it . This ensures it does not break when other upstreams are used.

Reply:
I suggest to have a unified interface for "add_adapter". Then in that interface (a function or a class), you can have apply corresponding adapters.

Reply:
added a unified adapter function , within the function its checks for upstream type and then applied the corresponding model operation